README.md

```diff
@@ -5,6 +5,15 @@
 
 This is a client library for interacting with the idam application.
 
+## Note
+
+This repository is not supported by the CFT IDAM team.
+
+IDAM is an OpenId Connect (OIDC) compliant authentication service, and any OIDC solution should be suitable for integrating with IDAM. All services that use IDAM are expected to use the OIDC endpoints for authentication. The IDAM integration guide is available here:
+https://tools.hmcts.net/confluence/display/SISM/OIDC+Integration+Guide
```

Comment:
The paragraph about OIDC is clearly about authentication. I agree that adding documentation about the purpose of this library would be useful for avoiding confusion.

Reply:
> The paragraph about OIDC is clearly about authentication

I don't really take your point.

Reply:
Well, you could have headings:

- OIDC User Authentication: Services, can use standard spring, not a key use of this library. Could contain the paragraph I suggested that links to the IDAM documentation.
- OIDC Password/client grants: A reason to use this library (could include details about length of time tokens are valid for, use of caching etc.)
- Search: A reason to use this library
- Tactical Auth: (authenticateUser, pins etc.) deprecated (or soon to be deprecated)

Reply:
It would make sense to add something like this, is that something you are up to doing?

> Use in tests: Not sure if that's something the library was intended for.

It wasn't written for that, but support was added for it and I think it's fairly widely used.

> Search: A reason to use this library (although I'm not sure it's in use outside of tests, and I'd prefer it to include something about paging if it were more widely used)

Heh a benefit of looking after the library is things like that you would take care of when writing :D, whomever wrote the code may not have been aware of that or was just doing the minimum they needed.

Reply:
Yeah, I'll add roughly those headings and update.